### DISCUSSION TITLE
Symfony 4.4 project fails to install as as generated Cache component is not compatible with it's bases class.

### DISCUSSION THREAD

**Discussion by hitsuji** [symfony/symfony#43020] [Q&A]
Hi Folks,
wondering if anyone else is experiencing this issue as well?

Trying to releaee some bug fixes for my 4.4 app and upon performing composer install I get thrown an error during the cache::clear.

`PHP Fatal error:  Declaration of ContainerNZ0v1dW\srcApp_KernelProdContainer::getParameter(string $name) must be compatible with Symfony\Component\DependencyInjection\Container::getParameter($name)`

The composer require block contains:

```
        "php": "^7.1.3",
        "ext-ctype": "*",
        "ext-iconv": "*",
        "guzzlehttp/guzzle": "~6.0",
        "incenteev/composer-parameter-handler": "^2.1",
        "sensio/framework-extra-bundle": "^5.5",
        "symfony/console": "4.4.*",
        "symfony/dotenv": "4.4.*",
        "symfony/flex": "^1.3.1",
        "symfony/framework-bundle": "4.4.*",
        "symfony/monolog-bundle": "^3.1",
        "symfony/process": "4.4.*",
        "symfony/security-bundle": "4.4.*",
        "symfony/serializer-pack": "*",
        "symfony/translation": "4.4.*",
        "symfony/twig-pack": "*",
        "symfony/yaml": "4.4.*"
```

**Comment by stof** [comment]
Please provide the output of `composer show 'symfony/*'`

**Comment by stof** [comment]
Note that if this happens when updating `symfony/dependency-injection` from 4.4 to 5.3 with an existing cache folder, a solution is to delete the existing cache generated with the old version.
but this looks more like a cache generated with `symfony/dependency-injection` 5.3 and loaded with 4.4, which is indeed not a supported case (we don't try to supporting downgrading packages in a BC way)

**Reply by hitsuji** [reply]
Ok, composer show 'symfony/*' shows a mix of 4.4 and 5.3

just noticed this aswell now that I try to redeploy the app again: Root composer.json requires symfony/property-access 4.4.*, found symfony/property-access[v4.4.0, ..., v4.4.30] but the package is fixed to v5.3.7

I've not commited my composer.lock so it looks like one of the deps i've used is forcing some deps to 5.3 versions.
I'll commit the composer.json from the previous release to fix the issue.
